pkgs/development/python-modules/detectron2/default.nix

```diff
@@ -79,14 +86,72 @@ buildPythonPackage {
     iopath
     omegaconf
     hydra-core
-    black
     packaging
-  ] ++ optional-dependencies.all;
+    black
+    torch # not explicitly declared in setup.py because they expect you to install it yourself
+    pydot # no idea why this is not in their setup.py
+  ];
 
   passthru.optional-dependencies = optional-dependencies;
 
-  # disable the tests for now until someone can check on a linux machine.
-  # doCheck = false;
+  nativeCheckInputs = [
+    av
+    opencv4
+    pytest-mock
+    pytestCheckHook
+    torchvision
+  ];
+
+  preCheck = ''
+    # prevent import errors for C extension modules
+    rm -r detectron2
+  '';
+
+  pytestFlagsArray = [
+    # prevent include $sourceRoot/projects/*/tests
+    "tests"
+  ];
+
+  disabledTestPaths = [
+    # try import caffe2
+    "tests/test_export_torchscript.py"
+    "tests/test_model_analysis.py"
+    "tests/modeling/test_backbone.py"
+    "tests/modeling/test_roi_heads.py"
+    "tests/modeling/test_rpn.py"
+    "tests/structures/test_instances.py"
+    # hangs for some reason
+    "tests/modeling/test_model_e2e.py"
+  ];
+
+  disabledTests = [
+    # failes for some reason
```

Comment:
```suggestion
    # fails for some reason
```
sorry for my typo.

Reply:
all good!
I just tested on my machine and I had one more test error
```
 E           TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).
```
that was in `test_map_style`
I'm disabling this and testing again.